From 2fccc9fdec0d55a75e1bcc6b8a256f38962d8946 Mon Sep 17 00:00:00 2001
From: David Bremner <david@tethera.net>
Date: Fri, 27 Jul 2018 17:26:14 +0800
Subject: [PATCH 1/3] debcherry fixup patch

d2efbdc make pkg-config substitutable for cross compilation
	 - no changes against upstream or conflicts
6f406e5 use lua5.2 on debian
	 - extra changes or conflicts
8956e2e pass LDFLAGS when compiling python and perl extension libs.
	 - extra changes or conflicts
f93cafb examples/swig/makefile: add on to CFLAGS rather than overriding them.
	 - extra changes or conflicts
dbd1a64 Set debian specific name for lua package.
	 - extra changes or conflicts
f8fd5f3 partial revert of 16f2ae68dc.  Modification of upstream makefile is no longer required to suppress a lintian warning.
	 - extra changes or conflicts
d460c96 Imported Upstream version 2.10
	 - extra changes or conflicts
3b29289 compile swig wrapper with -fPIC
	 - extra changes or conflicts
a0f117c install swig generated perl bindings, and an example script
	 - extra changes or conflicts
---
 extras/swig/makefile | 13 +++++++------
 src/makefile         |  7 ++++---
 2 files changed, 11 insertions(+), 9 deletions(-)

diff --git a/extras/swig/makefile b/extras/swig/makefile
index acc32dc..b5af4cd 100644
--- a/extras/swig/makefile
+++ b/extras/swig/makefile
@@ -1,13 +1,14 @@
 
 CXX=g++
-CFLAGS=-g -O2 -fPIC
+PKG_CONFIG ?= pkg-config
+CFLAGS+= -fPIC
 
 HL_INC=../../src/include/
 HL_SRC=../../src/
 
-LUA_CFLAGS=$(shell pkg-config --cflags lua)
-LUA_LIBS=$(shell pkg-config --libs lua)
-PYTHON_CFLAGS=$(shell pkg-config --cflags python-2.7)
+LUA_CFLAGS=$(shell $(PKG_CONFIG) --cflags lua5.2)
+LUA_LIBS=$(shell $(PKG_CONFIG) --libs lua5.2)
+PYTHON_CFLAGS=$(shell $(PKG_CONFIG) --cflags python-2.7)
 PERL_INC=$(shell perl -MExtUtils::Embed -eccopts)
 PHP_INC=$(shell php-config --includes)
 
@@ -19,12 +20,12 @@ lib-stamp:
 python: lib-stamp
 	swig -c++ -python -o highlight_wrap.cpp  highlight.i
 	${CXX} ${CFLAGS} -c highlight_wrap.cpp ${LUA_CFLAGS} ${PYTHON_CFLAGS} -I${HL_INC} 
-	${CXX} -shared -s highlight_wrap.o  -L${HL_SRC} -lhighlight ${LUA_LIBS} -lpython2.7 -o _highlight.so
+	${CXX} ${LDFLAGS} -shared -s highlight_wrap.o  -L${HL_SRC} -lhighlight ${LUA_LIBS} -lpython2.7 -o _highlight.so
 
 perl: lib-stamp
 	swig -c++ -perl -o highlight_wrap.cpp  highlight.i
 	${CXX} ${CFLAGS} -c highlight_wrap.cpp ${LUA_CFLAGS} ${PERL_INC} -I${HL_INC} 
-	${CXX} -shared -s highlight_wrap.o  -L${HL_SRC} -lhighlight ${LUA_LIBS} -o highlight.so
+	${CXX} ${LDFLAGS} -shared -s highlight_wrap.o  -L${HL_SRC} -lhighlight ${LUA_LIBS} -o highlight.so
 
 php: lib-stamp
 	swig -c++ -php -o highlight_wrap.cpp highlight.i
diff --git a/src/makefile b/src/makefile
index 20c5a80..df4adf1 100644
--- a/src/makefile
+++ b/src/makefile
@@ -14,6 +14,7 @@
 
 #CXX=clang++
 CXX=g++
+PKG_CONFIG ?= pkg-config
 
 # was: qmake-qt5
 QMAKE=qmake
@@ -39,13 +40,13 @@ INCLUDE_DIR=./include/
 # If Lua 5.2 is not default on your system yet you have to
 # use lua5.1 in the pkg-config calls
 
-LUA_CFLAGS=$(shell pkg-config --cflags lua)
+LUA_CFLAGS=$(shell $(PKG_CONFIG) --cflags lua5.2) -DUSE_LUA52
 
 # default lua lib
-LUA_LIBS=$(shell pkg-config --libs lua)
+LUA_LIBS=$(shell $(PKG_CONFIG) --libs lua5.2)
 
 # luajit lib
-# LUA_LIBS=$(shell pkg-config --libs luajit)
+# LUA_LIBS=$(shell $(PKG_CONFIG) --libs luajit)
 
 # Third-Party software paths
 ASTYLE_DIR=${CORE_DIR}astyle/
